src/main/java/com/google/api/generator/gapic/composer/grpc/GrpcServiceStubClassComposer.java

```diff
@@ -335,4 +447,120 @@ private LambdaExpr createRequestParamsExtractorClassInstance(Method method) {
         .setReturnExpr(returnExpr)
         .build();
   }
+
+  private void addRequestParamsForRoutingHeders(
+      Method method,
+      List<Statement> classStatements,
+      VariableExpr paramsVarExpr,
+      List<Expr> bodyExprs,
+      VariableExpr requestVarExpr) {
+    ImmutableList<RoutingHeader> routingHeaders = method.routingHeaders().routingHeadersList();
+    for (int i = 0; i < routingHeaders.size(); i++) {
+      RoutingHeader routingHeader = routingHeaders.get(i);
+      MethodInvocationExpr requestFieldGetterExpr =
+          createRequestFieldGetterExpr(requestVarExpr, routingHeader.field());
+      Expr routingHeaderKeyExpr =
+          ValueExpr.withValue(StringObjectValue.withValue(routingHeader.name()));
+      // TODO: Create proper snake style upper case name
+      String pathTemplateName =
+          String.format("PATH_TEMPLATE_%s_%s", method.name().toUpperCase(), i);
+      TypeNode pathTemplateType =
+          TypeNode.withReference(ConcreteReference.withClazz(PathTemplate.class));
+      Variable pathTemplateVar =
+          Variable.builder().setType(pathTemplateType).setName(pathTemplateName).build();
+      Expr routingHeaderPatternExpr = VariableExpr.withVariable(pathTemplateVar);
+      VariableExpr pathTemplateVarExpr =
+          VariableExpr.builder()
+              .setVariable(pathTemplateVar)
+              .setIsDecl(true)
+              .setIsStatic(true)
+              .setIsFinal(true)
+              .setScope(ScopeNode.PRIVATE)
+              .build();
+      ValueExpr valueExpr =
+          ValueExpr.withValue(StringObjectValue.withValue(routingHeader.pattern()));
+      Expr pathTemplateExpr =
+          AssignmentExpr.builder()
+              .setVariableExpr(pathTemplateVarExpr)
+              .setValueExpr(
+                  MethodInvocationExpr.builder()
+                      .setStaticReferenceType(pathTemplateType)
+                      .setMethodName("create")
+                      .setArguments(valueExpr)
+                      .setReturnType(pathTemplateType)
+                      .build())
+              .build();
+      Statement pathTemplateClassVar = ExprStatement.withExpr(pathTemplateExpr);
+      classStatements.add(pathTemplateClassVar);
+      MethodInvocationExpr addParamsMethodExpr =
+          MethodInvocationExpr.builder()
+              .setStaticReferenceType(
+                  TypeNode.withReference(ConcreteReference.withClazz(RoutingHeaderHelper.class)))
+              .setMethodName("addParams")
+              .setArguments(
+                  paramsVarExpr,
+                  requestFieldGetterExpr,
+                  routingHeaderKeyExpr,
+                  routingHeaderPatternExpr)
+              .build();
+
+      bodyExprs.add(addParamsMethodExpr);
+    }
+  }
+
+  private void addRequestParamsForHttpBindings(
```

Comment:
If i remember correctly, this logic is used heavily for rest transport and it does not have proper tests in this codebase yet (i.e. in other words it is very easy to break it). Please be very careful with this change. Ideally generate compute client and make sure it still works. (at least  unit tests pass).

Reply:
This class is in Grpc package which I assume it's only used for grpc requests not rest? I don't see any usage of this class for generating Http client either.